### PR TITLE
Add ci step to check for go vulnerabilities

### PR DESCRIPTION
**Pull request by sergiught**
### 📝 Checklist

- [x] All new/changed/fixed functionality is covered by tests (or N/A)
- [x] I have added documentation for all new/changed functionality (or N/A)

### CODE REVIEW

File: .github/workflows/lint.yaml [hunk 1]
```diff
@@ -13,13 +13,18 @@ jobs:
     runs-on: ubuntu-latest
     steps:
       - name: install go
-        uses: actions/setup-go@v2
+        uses: actions/setup-go@v3
         with:
-          go-version: 1.17
```
Comment (sergiught) [inline .github/workflows/lint.yaml:18]: We need at least 1.18 in the CI to install and run go vuln check.

Comment (Widcket) [reply .github/workflows/lint.yaml:18]: Should we use go 1.19 instead?

Comment (sergiught) [reply .github/workflows/lint.yaml:18]: IMO we'd stay as close as possible to the minimum go version expressed in the go.mod https://github.com/auth0/go-jwt-middleware/blob/e00e3d716644faff0502846652343c5f40d4e1d2/go.mod#L3

Also I wouldn't bump the version in the go mod yet as 1.17 is still fully supported by google (they support the last 2 released versions before the latest, e.g. right now we are on 1.19 so 1.18 and 1.17 are still supported).

**Review by Widcket — APPROVED** [review]

File: .github/workflows/lint.yaml (hunk 1 above)
Comment (Widcket) [reply .github/workflows/lint.yaml:18]: Consider having a matrix for building in all supported Go versions (not linting/checking/etc, just building).

**Pull request merged**